SI/Sem.09/Pract.09/Task06.cpp

```diff
@@ -0,0 +1,24 @@
+﻿#include <iostream>
+const int MAX_SIZE = 20;
+void addElement(int* array, int toAdd, int len)
+{
+	if (len == 0 || toAdd > array[len - 1])
+	{
+		array[len] = toAdd;
```

Comment:
We need to resize here.
You have the MAX_SIZE.
array[len] is INVALID

Reply:
void resize(int*& array, int& size) {
	int* newArray = new int[size * 2];
	for (int i = 0; i < size; i++) {
		newArray[i] = array[i];
	}
	size *= 2;
	delete[] array;
	array = newArray;
}

void addElement(int* array, int toAdd, int len)
{
	resize(array, len);
	if (len == 0 || toAdd > array[len - 1])
	{
		array[len] = toAdd;
		return;
	}
	array[len] = array[len - 1];
	addElement(array, toAdd, len - 1);
}

Reply:
@AtanasHrist0v will take over for task06 with an example solution.

Reply:
Link: https://pastebin.com/2b1b3sE1?fbclid=IwAR2vAJXMvd5n-I9lyYZN1ff9wGDraNjnKAJtTVYy7RFHAbX3qeuoLCs1J-c